### ISSUE TITLE
Upgrading to webpack 4 issue 

### ISSUE THREAD

**Issue by cpBurn** [jantimon/favicons-webpack-plugin#126]
I'm getting the following when trying to build webpack 4:

TypeError: Cannot read property 'tapAsync' of undefined
    at /Users/user/react-boilerplate/node_modules/favicons-webpack-plugin/index.js:73:62

this is the code in `index.js:73`

```
if (!tapped++) {
   cmpp.hooks.htmlWebpackPluginBeforeHtmlProcessing.tapAsync(
       'favicons-webpack-plugin',
       addFaviconsToHtml
    );
}
```

any ideas?

**Comment by aldarund** [comment]
Same issue.

**Comment by aldarund** [comment]
I have created pr that fixes this -> https://github.com/jantimon/favicons-webpack-plugin/pull/127

**Comment by cpBurn** [comment]
Thanks a lot @aldarund, would be awesome to get this release :)

**Comment by szykov** [comment]
Any progress regarding webpack 4 support?

**Comment by Greg808** [comment]
Same Issue. Any progress regarding webpack 4 support?

**Comment by matiishyn** [comment]
how did you fix it @Greg808 ?

**Comment by Greg808** [comment]
actually is used this plugin https://github.com/brunocodutra/webapp-webpack-plugin witch is a fork of your plugin and brunocodutra fixed it

**Comment by szykov** [comment]
generating icons can be done manually one time. no sense to put it in webpack scripts and generate them every time you bundle your scripts.

**Comment by ConneXNL** [comment]
Still getting this issue...

**Comment by grigory-leonenko** [comment]
Issue still present

**Comment by perevezencev** [comment]
+1

**Comment by missmatsuko** [comment]
`webapp-webpack-plugin` didn't work for my project, but I found someone made a fork of this package for Webpack 4 that looks promising: https://github.com/romanlex/app-manifest-webpack-plugin

**Comment by maxpaj** [comment]
Ran into this issue today.

**Comment by danial24** [comment]
Ran into the same issue today

**Comment by zlv-thisF** [comment]
+1

**Comment by zeravcic** [comment]
It seems that solution is to add **mandatory** `html-webpack-plugin` plugin! Yes?

## Environment
- JeffreyWay/laravel-mix [v4.0.16](https://github.com/JeffreyWay/laravel-mix/tree/d5bc244dafad62464f7404ec9851e087c209d8ae)
- jantimon/html-webpack-plugin [v3.2.0](https://github.com/jantimon/html-webpack-plugin/tree/v3.2.0) 
- jantimon/favicons-webpack-plugin [v0.0.9](https://github.com/jantimon/favicons-webpack-plugin/commit/7845e4f20a54776555674ecc0537ca3d89aa2ccd) - commit [7845e4f20a54776555674ecc0537ca3d89aa2ccd]

## Possible solution
This solution works for me (using [Laravel Mix v4.0.16](https://www.npmjs.com/package/laravel-mix/v/4.0.16))
Code:
```js
const mix = require('laravel-mix')
const HtmlWebpackPlugin = require('html-webpack-plugin')
const FaviconsWebpackPlugin = require('favicons-webpack-plugin')

const htmlWebpackPlugin =  new HtmlWebpackPlugin();
const faviconsWebpackPlugin = new FaviconsWebpackPlugin({
    logo: path.join('path/to/my-logo.png'),
    prefix: 'path/to/favicons-[hash]/',
    emitStats: true,
    statsFilename: 'path/to/faviconstats.json',
});

/*
 |--------------------------------------------------------------------------
 | Webpack config
 |--------------------------------------------------------------------------
 */
mix.webpackConfig({
    plugins: [
        htmlWebpackPlugin,
        faviconsWebpackPlugin,
    ]
})
```

...and then of course - `$ npm run production`

I am waiting to hear some others experiences :) 👍 👎

**Comment by jantimon** [comment]
Solved in #157 please try favicons-webpack-plugin 1.0.0 and let me know if it doesn't work for you
